specification/hybridcompute/resource-manager/Microsoft.HybridCompute/preview/2022-08-11-preview/HybridCompute.json

```diff
@@ -0,0 +1,1637 @@
+        },
+        "startupType": {
+          "type": "string",
+          "description": "The behavior of the service when the Arc-enabled machine starts up."
+        }
+      },
+      "description": "Describes the status and behavior of a service."
+    },
+    "CloudMetadata": {
+      "type": "object",
+      "properties": {
+        "provider": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Specifies the cloud provider (Azure/AWS/GCP...)."
+        }
+      },
+      "description": "The metadata of the cloud environment (Azure/GCP/AWS/OCI...)."
+    },
+    "OSProfile": {
+      "type": "object",
+      "properties": {
+        "computerName": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Specifies the host OS name of the hybrid machine."
+        },
+        "windowsConfiguration": {
+          "type": "object",
+          "properties": {
+            "patchSettings": {
+              "x-ms-client-flatten": true,
+              "$ref": "#/definitions/PatchSettings"
+            }
+          },
+          "description": "Specifies the windows configuration for update management."
+        },
+        "linuxConfiguration": {
+          "type": "object",
+          "properties": {
+            "patchSettings": {
+              "x-ms-client-flatten": true,
+              "$ref": "#/definitions/PatchSettings"
+            }
+          },
+          "description": "Specifies the linux configuration for update management."
+        }
+      },
+      "description": "Specifies the operating system settings for the hybrid machine."
+    },
+    "DetectedProperties": {
+      "readOnly": true,
+      "type": "object",
+      "additionalProperties": {
+        "type": "string"
+      },
+      "description": "Detected properties from the machine."
+    },
+    "MachineProperties": {
+      "type": "object",
+      "properties": {
+        "locationData": {
+          "$ref": "../../../../../common-types/resource-management/v1/types.json#/definitions/locationData"
+        },
+        "agentConfiguration": {
+          "$ref": "#/definitions/AgentConfiguration",
+          "description": "Configurable properties that the user can set locally via the azcmagent config command, or remotely via ARM."
+        },
+        "serviceStatuses": {
+          "$ref": "#/definitions/ServiceStatuses",
+          "description": "Statuses of dependent services that are reported back to ARM."
+        },
+        "cloudMetadata": {
+          "$ref": "#/definitions/CloudMetadata",
+          "description": "The metadata of the cloud environment (Azure/GCP/AWS/OCI...)."
+        },
+        "osProfile": {
+          "$ref": "#/definitions/OSProfile",
+          "description": "Specifies the operating system settings for the hybrid machine."
+        },
+        "provisioningState": {
+          "readOnly": true,
+          "type": "string",
```

Comment:
>string [](http://example.com/codeflow?start=19&length=6)

should be an enum with the min values - Canceled, Failed and Succeeded

Reply:
Hi @rkmanda. These are not related to our current set of updates to the spec.

We can take these as items for our next version update.

Currently delayed due to the slow progress of this spec review.

Reply:
Hi @rkmanda. These are not related to our current set of updates to the spec.

We can take these as items for our next version update.

Currently delayed due to the slow progress of this spec review.

Reply:
@rkmanda, breaking change review board counts this kind of change as a breaking change and against us to change from string to enum sets